<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1287</width>
    <height>933</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="4" column="1">
     <widget class="QPushButton" name="rotateYbutton">
      <property name="text">
       <string>Rotate Y</string>
      </property>
     </widget>
    </item>
    <item row="26" column="1">
     <widget class="QPushButton" name="vertexColorButton">
      <property name="text">
       <string>Цвет вершин</string>
      </property>
     </widget>
    </item>
    <item row="15" column="1">
     <widget class="QPushButton" name="bgColorButton">
      <property name="text">
       <string>Цвет фона</string>
      </property>
     </widget>
    </item>
    <item row="19" column="1">
     <widget class="QComboBox" name="edgeTypeCombo">
      <item>
       <property name="text">
        <string>Сплошные</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Пунктирные</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="27" column="1">
     <spacer name="verticalSpacer_5">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="0" rowspan="43">
     <widget class="s21::GLWidget" name="openGLWidget"/>
    </item>
    <item row="32" column="1">
     <widget class="QPushButton" name="resetSettingsButton">
      <property name="text">
       <string>Сброс настроек</string>
      </property>
     </widget>
    </item>
    <item row="20" column="1">
     <spacer name="verticalSpacer_9">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="30" column="1">
     <widget class="QComboBox" name="projectionCombo">
      <item>
       <property name="text">
        <string>Перспективная</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Ортографическая</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QPushButton" name="rotateXbutton">
      <property name="text">
       <string>Rotate X</string>
      </property>
     </widget>
    </item>
    <item row="29" column="1">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Проекция:</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <spacer name="verticalSpacer_8">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="12" column="1">
     <widget class="QDoubleSpinBox" name="moveZSpin">
      <property name="minimum">
       <double>-100.000000000000000</double>
      </property>
      <property name="maximum">
       <double>100.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="13" column="1">
     <widget class="QPushButton" name="moveApplyButton">
      <property name="text">
       <string>Move</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Policy::Expanding</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="14" column="1">
     <spacer name="verticalSpacer_6">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="39" column="1">
     <widget class="QLabel" name="statusLabel">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="text">
       <string>Имя файла .obj: </string>
      </property>
      <property name="textFormat">
       <enum>Qt::TextFormat::RichText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="16" column="1">
     <widget class="QPushButton" name="edgeColorButton">
      <property name="text">
       <string>Цвет ребер</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="pushButton_2">
      <property name="text">
       <string>Открыть .obj</string>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QPushButton" name="rotateZbutton">
      <property name="text">
       <string>Rotate Z</string>
      </property>
     </widget>
    </item>
    <item row="18" column="1">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Тип рёбер</string>
      </property>
     </widget>
    </item>
    <item row="24" column="1">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Размер вершин:</string>
      </property>
     </widget>
    </item>
    <item row="10" column="1">
     <widget class="QDoubleSpinBox" name="moveXSpin">
      <property name="minimum">
       <double>-100.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
      <property name="value">
       <double>0.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="40" column="1">
     <spacer name="verticalSpacer_4">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="7" column="1">
     <widget class="QPushButton" name="zoomInButton">
      <property name="text">
       <string>Zoom +</string>
      </property>
     </widget>
    </item>
    <item row="38" column="1">
     <widget class="QPushButton" name="pushButton">
      <property name="maximumSize">
       <size>
        <width>100</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>Выход</string>
      </property>
     </widget>
    </item>
    <item row="33" column="1">
     <spacer name="verticalSpacer_7">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="25" column="1">
     <widget class="QDoubleSpinBox" name="vertexSizeSpin">
      <property name="decimals">
       <number>0</number>
      </property>
      <property name="minimum">
       <double>2.000000000000000</double>
      </property>
      <property name="maximum">
       <double>20.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>2.000000000000000</double>
      </property>
      <property name="value">
       <double>2.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="8" column="1">
     <widget class="QPushButton" name="zoomOutButton">
      <property name="text">
       <string>Zoom -</string>
      </property>
     </widget>
    </item>
    <item row="22" column="1">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Тип вершин:</string>
      </property>
     </widget>
    </item>
    <item row="9" column="1">
     <spacer name="verticalSpacer_3">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="37" column="1">
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="31" column="1">
     <widget class="QPushButton" name="resetButton">
      <property name="text">
       <string>Сброс модели</string>
      </property>
     </widget>
    </item>
    <item row="23" column="1">
     <widget class="QComboBox" name="vertexTypeCombo">
      <item>
       <property name="text">
        <string>Нет</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Круг</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Квадрат</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="17" column="1">
     <widget class="QDoubleSpinBox" name="edgeWidthSpin">
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>1.000000000000000</double>
      </property>
      <property name="maximum">
       <double>10.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="11" column="1">
     <widget class="QDoubleSpinBox" name="moveYSpin">
      <property name="minimum">
       <double>-100.000000000000000</double>
      </property>
      <property name="maximum">
       <double>100.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="36" column="1">
     <widget class="QPushButton" name="gifRecordButton">
      <property name="text">
       <string>Записать GIF</string>
      </property>
     </widget>
    </item>
    <item row="34" column="1">
     <widget class="QPushButton" name="saveSnapshotButton">
      <property name="text">
       <string>Сохранить снимок</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>view/glwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1000</x>
     <y>466</y>
    </hint>
    <hint type="destinationlabel">
     <x>909</x>
     <y>472</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
